errorDisplay.js

```diff
@@ -2,109 +2,85 @@
-	function showWidget(errorText, lineStart){
-		var editor    = EditorManager.getActiveEditor();
-		var lineStats = editor._codeMirror.lineInfo(lineStart);
-
-		if(!lineStats.widgets && widgetsErrors.length === 0){
-            var htmlNode = document.createElement("div");
-            htmlNode.className = "errorPanel";
-            var text = Mustache.render(lineWidgetHTML, { "error": errorText });
-            htmlNode.innerHTML = text;
-
-        
-			var errrorWidget = editor._codeMirror.addLineWidget(lineStart, htmlNode,
-				{coverGutter: false, noHScroll: false, above: false, showIfHidden: false});
-			
-			widgetsErrors.push(errrorWidget);
-		}
-	}
-
-	//Function that removes the line widget (errors)
-	function removeWidget(){
-		//Remove displayed error messages
-		widgetsErrors.forEach(function (lineWidget) {
-			if (lineWidget) {
-				lineWidget.clear();
-			}
-		});
-		widgetsErrors = [];
-	}
-
-	//Function that adds a button on the gutter (on given line nubmer) next to the line numbers
-	function showGutter(lineStart){
-		if(gutters.length === 0){
-			var editor = EditorManager.getFocusedEditor();
-			var $errorDiv = $("<div class='error'/>");
-			var $errorMarker = $("<span class='errorButton'/>");
-			var foundGutters = ["errorButton"];
-
-			$errorDiv.append($errorMarker);
-			$errorMarker.addClass("errorText");
-			$errorMarker.text("!");
-
-			gutters.push(editor._codeMirror.setGutterMarker(lineStart, "errorButton", $errorDiv[0]));
-
-			editor._codeMirror.setOption("gutters", foundGutters);
-			//Show tooltips message
-			$(".CodeMirror-linenumbers").tooltipsy({content : "Click button for information"}); 
-
-			$(".CodeMirror-gutter").addClass("gutterCursor");
-		}
-	}
-
-	//Function that removes gutter button
-	function removeGutter(){
-		var editor = EditorManager.getFocusedEditor();
-		gutters = [];
-		editor._codeMirror.clearGutter("errorButton");
-		//Destroy tooltips instance
-		$(".CodeMirror-linenumbers").data("tooltipsy").destroy();
-		//Changes cursor back to default
-		$(".CodeMirror-gutter").removeClass("gutterCursor");
-	}
-	
-	exports.markErrors = markErrors;
-	exports.clearErrors = clearErrors;
-	exports.showWidget = showWidget;
-	exports.showGutter = showGutter;
-	exports.removeGutter = removeGutter;
-	exports.removeWidget = removeWidget;
-});
\ No newline at end of file
+    "use strict";
+
+    var EditorManager  = brackets.getModule("editor/EditorManager"),
+        ExtensionUtils = brackets.getModule("utils/ExtensionUtils"),
+        lineWidgetHTML = require("text!inlineWidget.html"),
+        currentErrorWidget;
+
+    ExtensionUtils.loadStyleSheet(module, "main.less");
+    require("tooltipsy.source");
+
+    function initGutter() {
+        EditorManager.getActiveEditor()._codeMirror.setOption("gutters", ["errors"]);
+    }
+
+    function highlight(line) {
+        if(!line) {
+           return;
+        }
+
+        var editor = EditorManager.getActiveEditor();
```

Comment:
Also add a function for getting the CodeMirror instance, since you do that a lot too.